### PR TITLE
Mail messages

### PR DESCRIPTION
**Pull request by bradfordcondon**
accomplish #13, #26 

please note:
* i alter the `tripal_hq_save_submission` API to return the submission ID rather than the transaction type.
* I havent tested each scenarios output mail yet, ill do that next, but want @almasaeed2010 's 👍 on changing his API

### CODE REVIEW

**Comment by bradfordcondon** [comment]
all messages confirmed to fire, see https://github.com/statonlab/tripal_hq/issues/2#issuecomment-430818897

File: includes/tripal_hq_user_data.form.inc [hunk 1]
```diff
@@ -219,20 +219,14 @@ function tripal_hq_entity_form_submit($form, &$form_state) {
...
 
-  $op = tripal_hq_save_submission($entity, $sid);
-  if ($op === 'inserted') {
+  $sid = tripal_hq_save_submission($entity, $sid);
 
-    //From site, to end user
-    drupal_mail('tripal_hq', 'user_submit_notice', $user_email, language_default(), $values, $site_email, TRUE);
-    //from site to site_admin
-    drupal_mail('tripal_hq', 'admin_submit_notice', $relevant_admins, language_default(), $values, $site_email, TRUE);
+  if ($sid) {
```
Comment (almasaeed2010) [inline includes/tripal_hq_user_data.form.inc]: This no longer makes sense. The else message should indicate an error. Currently, when an error happens it says the record "updated successfully".

File: includes/tripal_hq_user_data.form.inc [hunk 2]
```diff
@@ -251,7 +245,7 @@ function tripal_hq_entity_form_submit($form, &$form_state) {
  * @param $entity
  * @param null $submission_id
  *
- * @return string 'updated' or 'inserted'
+ * @return string submission id, or FALSE if error
```
Comment (almasaeed2010) [inline includes/tripal_hq_user_data.form.inc]:
```suggestion
 * @return int|boolean submission id, or FALSE if error
```

File: includes/tripal_hq_user_data.form.inc (hunk 1 above)
Comment (almasaeed2010) [reply includes/tripal_hq_user_data.form.inc]: So you want 2 checks here. The first, to make sure $sid is not false and thus no error happened. The second is
 ```php
if(isset($form_state['values']['submission_id'])) {
  // Tell the user that an update happened
} else {
  // Tell the user that a new submission was created.
}
```

**Pull request merged**